addons/stock/models/stock_picking.py

```diff
@@ -954,14 +961,18 @@ def _log_activity(self, render_method, documents):
         the render_method should return a string with an html format
         :param stream string:
         """
-        for (parent, responsible), rendering_context in documents.items():
+        for (parent, responsible, activity), rendering_context in documents.items():
             note = render_method(rendering_context)
-            parent.activity_schedule(
-                'mail.mail_activity_data_warning',
-                date.today(),
-                note=note,
-                user_id=responsible.id or SUPERUSER_ID
-            )
+            if activity:
+                parent.activity_schedule(
+                    'mail.mail_activity_data_warning',
+                    date.today(),
+                    note=note,
+                    user_id=responsible.id or SUPERUSER_ID
+                )
+            else:
+                order_id = parent
+                order_id.message_post(body=note)
```

Comment:
care to explain why there are cases you do a message_post? the specs only talk about activities afaik

Reply:
if any record is in draft stage, we can update it, in that case, we/system can directly make changes to the record, so in such type of cases, we need to just inform the user that, this kind of changes has been done to the record, but if the record is in confirm or any other stage, in which system cannot make any changes directly, in that case, we need to inform user with the help of activity that user need to change the document manually.